src/Acl.php

```diff
@@ -1024,11 +1024,24 @@ protected function getRuleType(
 
         // Was a custom assertion supplied? Use it to retrieve the rule type.
         if ($rule['assert']) {
-            return $this->getRuleTypeFromAssertion($rule['assert'], $rule['type'], $role, $resource);
+            $assertion = $rule['assert'];
+            $assertionValue = $assertion->assert(
+                $this,
+                ($this->isAllowedRole instanceof Role\RoleInterface) ? $this->isAllowedRole : $role,
+                ($this->isAllowedResource instanceof Resource\ResourceInterface) ? $this->isAllowedResource : $resource,
+                $this->isAllowedPrivilege
+            );
+        }
+
+        if (null === $rule['assert'] || $assertionValue) {
+            return $rule['type'];
+        } elseif (null !== $resource || null !== $role || null !== $privilege) {
+            return;
+        } elseif (self::TYPE_ALLOW === $rule['type']) {
+            return self::TYPE_DENY;
         }
```

Comment:
```suggestion
        if (null === $rule['assert'] || $assertionValue) {
            return $rule['type'];
        }
        
        if (null !== $resource || null !== $role || null !== $privilege) {
            return;
        }
        
        if (self::TYPE_ALLOW === $rule['type']) {
            return self::TYPE_DENY;
        }
```

There's no need for an `elseif` if you are returning from the block. Additionally, it makes it easier to identify the distinct exit conditions.